src/hooks/users/useRecentLocationUsers.ts

```diff
@@ -14,22 +21,37 @@ import { useWorldUsers } from "./useWorldUsers";
  * @example useRecentLocationUsers(venue.name)
  * @example useRecentLocationUsers(`${venue.name}/${roomTitle}`)
  */
-export const useRecentLocationUsers = (locationName?: string) => {
+export const useRecentLocationUsers = (
+  locationName?: string
+): {
+  recentLocationUsers: readonly WithId<User>[];
+  isRecentLocationUsersLoaded: boolean;
+} => {
   const lastSeenThreshold = useUserLastSeenThreshold();
-  const { worldUsers, isWorldUsersLoaded } = useWorldUsers();
-
-  return useMemo(
-    () => ({
-      recentLocationUsers: locationName
-        ? worldUsers.filter(
-            (user) =>
-              user.lastSeenIn?.[locationName] &&
-              normalizeTimestampToMilliseconds(user.lastSeenIn[locationName]) >
-                lastSeenThreshold
-          )
-        : [],
-      isRecentLocationUsersLoaded: isWorldUsersLoaded,
-    }),
-    [worldUsers, isWorldUsersLoaded, lastSeenThreshold, locationName]
-  );
+
+  useConnectWorldUsers();
+
+  const { recentLocationUsers, isWorldUsersLoaded } = useSelector((state) => {
+    const worldUsers = worldUsersSelector(state);
+    const isWorldUsersLoaded = isLoaded(worldUsers);
+
+    if (!worldUsers || !locationName)
+      return { recentLocationUsers: noUsers, isWorldUsersLoaded };
+
+    const recentLocationUsers = worldUsers
+      .filter(
+        (user) =>
+          user.lastSeenIn?.[locationName] &&
+          normalizeTimestampToMilliseconds(user.lastSeenIn[locationName]) >
+            lastSeenThreshold
+      )
+      .map(userWithLocationToUser);
+
+    return { recentLocationUsers, isWorldUsersLoaded };
+  }, isEqual);
```

Comment:
Since the firestore connect was removed, Do we have users in the redux state? Given that we handle it using the context

Reply:
I'd have to check my code again, but i'm pretty sure the firestore connect still exists within the context provider doesn't it?

Reply:
Yep, it does. It's just unconventional way of utilizing Context for our codebase so I had questions

Reply:
See further explanations of the 'why' of this in https://github.com/sparkletown/sparkle/pull/1703#discussion_r666014344

Reply:
Marking this as resolved. Feel free to reopen if needs be.